### ISSUE TITLE
Prevent creation of Ports with same VNI + MAC address

### ISSUE THREAD

**Issue by rzezeski** [oxidecomputer/opte#184]
MAC addresses within a given virtual network should be unique.

**Comment by rzezeski** [comment]
I tested this via `opteadm` by verifying it refused to create two Ports with the same VNI + MAC.

```
rpz@kalm:~/oxidecomputer/opte/xde$ pfexec opteadm create-xde opte104 --bsvc-addr "fd00:99::1" --bsvc-vni 999 --gateway-ip 172.30.0.1 --gateway-mac 'a8:40:25:f7:77:77' --private-ip 172.30.0.24 --private-mac 'a8:40:25:f7:ce:59' --src-underlay-addr 'fd00:1122:3344:101::1' --vpc-subnet 172.30.0.0/24 --vpc-vni 9072780

rpz@kalm:~/oxidecomputer/opte/xde$ pfexec opteadm list-ports
LINK                             MAC ADDRESS              IPv4 ADDRESS     STATE   
opte104                          A8:40:25:F7:CE:59        172.30.0.24      ready
   
rpz@kalm:~/oxidecomputer/opte/xde$ pfexec opteadm create-xde opte105 --bsvc-addr "fd00:99::1" --bsvc-vni 999 --gateway-ip 172.30.0.1 --gateway-mac 'a8:40:25:f7:77:77' --private-ip 172.30.0.25 --private-mac 'a8:40:25:f7:ce:59' --src-underlay-addr 'fd00:1122:3344:101::1' --vpc-subnet 172.30.0.0/24 --vpc-vni 9072780
ERROR: command CreateXde failed: MacExists { port: "opte105", vni: Vni { inner: 9072780 }, mac: MacAddr { inner: A8:40:25:F7:CE:59 } }
```

**Comment by rzezeski** [comment]
Addressed in fa58951b8b5336109622fd67b0668b070fcd8494.
